google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -331,7 +335,7 @@ public Builder setTimeStorageClassUpdatedOffsetDateTime(
     }
 
     /** Sets the blob's user provided metadata. */
-    public abstract Builder setMetadata(Map<String, String> metadata);
+    public abstract Builder setMetadata(@Nullable Map<@NonNull String, @Nullable String> metadata);
```

Comment:
If a dev is extended BlobInfo and they overload; they'll need to add @Nullable as well IIUC?

Reply:
The annotations are optional from the actual runtime standpoint, this is putting a label to the existing behavior that wasn't explicitly expressed anywhere.